src/wallet/scriptpubkeyman.cpp

```diff
@@ -1987,57 +1987,51 @@ std::optional<MigrationData> LegacyDataSPKM::MigrateToDescriptor()
         }
     }
 
-    // Multisigs are special. They don't show up as ISMINE_SPENDABLE unless they are in a P2SH
-    // So we have to check if any of our scripts are a multisig and if so, add the P2SH
-    for (const auto& script_pair : mapScripts) {
-        const CScript script = script_pair.second;
+    // Make sure that we have accounted for all scriptPubKeys
+    assert(spks.size() == 0);
+
+    // Legacy wallets can also contains scripts whose P2SH, P2WSH, or P2SH-P2WSH it is not watching for
+    // but can provide script data to a PSBT spending them. These "solvable" output scripts will need to
+    // be put into the separate "solvables" wallet.
+    // These can be detected by going through the entire candidate output scripts, finding the ISMINE_NO scripts,
+    // and checking CanProvide() which will dummy sign.
+    for (const CScript& script : GetCandidateScriptPubKeys()) {
+        // Since we only care about P2SH, P2WSH, and P2SH-P2WSH, filter out any scripts that are not those
+        if (!script.IsPayToScriptHash() && !script.IsPayToWitnessScriptHash()) {
+            continue;
+        }
+        if (IsMine(script) != ISMINE_NO) {
+            continue;
+        }
+        SignatureData dummy_sigdata;
+        if (!CanProvide(script, dummy_sigdata)) {
+            continue;
+        }
 
         // Get birthdate from script meta
         uint64_t creation_time = 0;
-        const auto& it = m_script_metadata.find(CScriptID(script));
-        if (it != m_script_metadata.end()) {
-            creation_time = it->second.nCreateTime;
+        const auto& mit = m_script_metadata.find(CScriptID(script));
+        if (mit != m_script_metadata.end()) {
+            creation_time = mit->second.nCreateTime;
         }
 
-        std::vector<std::vector<unsigned char>> sols;
-        TxoutType type = Solver(script, sols);
-        if (type == TxoutType::MULTISIG) {
-            CScript sh_spk = GetScriptForDestination(ScriptHash(script));
-            CTxDestination witdest = WitnessV0ScriptHash(script);
-            CScript witprog = GetScriptForDestination(witdest);
-            CScript sh_wsh_spk = GetScriptForDestination(ScriptHash(witprog));
-
-            // We only want the multisigs that we have not already seen, i.e. they are not watchonly and not spendable
```

Comment:
ca0564e02c72ea57d89ec37f0f4948e6a5d0096e: I think it's worth preserving some of this comment, if only to explain to future readers why we went for the `GetCandidateScriptPubKeys() | CanProvide` approach.

Perhaps in `CanProvide`?

Reply:
I don't think any part of this comment is relevant to the current code.